cube/src/main/java/zone/cogni/asquare/cube/spel/TemplateService.java

```diff
@@ -0,0 +1,23 @@
+package zone.cogni.asquare.cube.spel;
+
+import org.springframework.core.io.Resource;
+
+public interface TemplateService {
+
+    String processTemplate(String template, Object root);
+
+    String processTemplate(Resource templateResource, Object root);
+
+    default NamedTemplate processTemplate(NamedTemplate namedTemplate, Object root) {
+        String result = processTemplate(namedTemplate.getTemplate(), root);
+        return getNewNamedTemplate(namedTemplate, root, result);
+    }
+
+    default NamedTemplate getNewNamedTemplate(NamedTemplate namedTemplate, Object root, String result) {
```

Comment:
this was a private method but java 8 does not allow this. maybe `getNamedTemplateCopy` is more clear?